shell.nix

```diff
@@ -3,6 +3,7 @@
 
 pkgs.mkShell {
   buildInputs = with pkgs.python3Packages; [
```

Comment:
It's a pythonApplication in nix, not a pythonPackage, so it's in pkgs.
```nix
  buildInputs = (with pkgs; [
    editorconfig-checker
  ])
  ++ (with pkgs.python3Packages; [
    mkdocs
    mkdocs-material
    mkdocs-git-revision-date-localized-plugin
  ]);
```

Reply:
I'm not a nix-shell users. Is it looking how you had hoped now? If no, maybe you should make the changes because I'm not familiar with nix-shell.